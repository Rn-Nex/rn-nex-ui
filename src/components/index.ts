export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
