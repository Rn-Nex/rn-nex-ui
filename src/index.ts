export * from './components';
export * from './libraries';
export * from './utils';
