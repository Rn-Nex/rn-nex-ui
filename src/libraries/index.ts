export * from './themes/v2';
export * from './style';
