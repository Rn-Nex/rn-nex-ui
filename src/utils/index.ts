export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
