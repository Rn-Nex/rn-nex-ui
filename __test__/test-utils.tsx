import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
