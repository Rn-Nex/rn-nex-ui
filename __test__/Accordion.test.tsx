import React from 'react';
import { render } from './test-utils';
import { Accordion, AccordionSummary, Text } from '../src';
import { View } from 'react-native';

describe('Accordion Component', () => {
  const mockAccordionTestId = 'accordion-test-id';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<Accordion />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<Accordion ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should change the border radius when passed the square prop', () => {
    const { getByTestId } = render(<Accordion testID={mockAccordionTestId} square />);

    const accordion = getByTestId(mockAccordionTestId);
    expect(accordion.props.style).not.toHaveProperty('borderRadius');
  });

  it('should change the opacity when passed the disable prop', () => {
    const { getByTestId } = render(<Accordion testID={mockAccordionTestId} disable />);

    const accordion = getByTestId(mockAccordionTestId);
    expect(accordion.props.style).toEqual(expect.objectContaining({ opacity: 0.5 }));
  });
});

describe('AccordionSummary Component', () => {
  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<AccordionSummary />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<AccordionSummary ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should render the icon when passed the expandIcon prop', () => {
    const { getByText } = render(<AccordionSummary expandIcon={<Text>expandIcon</Text>} />);

    const expandIcon = getByText('expandIcon');
    expect(expandIcon).toBeDefined();
  });
});
