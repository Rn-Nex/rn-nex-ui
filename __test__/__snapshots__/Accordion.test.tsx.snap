// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component should render correctly 1`] = `[Function]`;

exports[`AccordionSummary Component should render correctly 1`] = `[Function]`;
